src/services/completions.ts

```diff
@@ -335,6 +322,50 @@ namespace ts.Completions {
         }
     }
 
+    function getJsxClosingTagCompletion(location: Node | undefined, sourceFile: SourceFile): CompletionInfo | undefined {
+        // When there is a full closing tag and the tag includes a property access expression,
+        // the node hierarchy to get the JSX closing element is at grand parent of location.
+        // For example:
+        //   var x = <Comp.Item></Comp.Item/*1*/
+        //   var y = <Comp.Item></Comp.Item/*2*/>
+        const jsxClosingElement = location && location.parent
+            ? location.parent.parent && isJsxClosingElement(location.parent.parent)
+                ? location.parent.parent
+                : isJsxClosingElement(location.parent)
+                    ? location.parent
+                    : undefined
+            : undefined;

```

Comment:
This will only work for a fixed depth of property access expression, no? Add another dot and `location.parent.parent` will still be in the property access expression. You want to do a loop, continuing for as long as you’re on a PropertyAccessExpression, quitting with success if you hit a `JsxClosingElement`, quitting with failure if you hit anything else. You can use the `findAncestor` utility for this.

Reply:
Hah, I just realized this is why your `x7` and `x8` tests are only `Exp.M`, not `Exp.M.SFCComp`. Make those tests the whole thing, then they’ll fail until you fix this!

Reply:
As far as I tested if we have any kinda expression the parent will be the closing tag, and there was the only other case when we write the full closing tag name including property access expression the grand parent was the closing tag.

I'm not sure if there is any case that we have to dig higher to find the closing tag. And on top of that what would happen if we go to many ancestors high considering this completion can be asked at any location.

Reply:
At `</ /**/Foo.Bar.Blah.Baz.Wut>`, the closing tag is `location.parent.parent.parent.parent`!

Reply:
> what would happen if we go to many ancestors high

This is why you bail if you hit something that’s neither `PropertyAccessExpression` nor `JsxClosingElement`.

Reply:
> This is why you bail if you hit something that’s neither `PropertyAccessExpression` nor `JsxClosingElement`.

Oh is that already part of `findAncestor()` implementation? cool I'll look into that and gonna update the PR

Reply:
I just added that test-case you mentioned and tested with current implementation and works 🤷‍♂️

Reply:
Maybe if we would have something like `</ /*1*/ Foo.Bar  Baz.Wut   >` it can happen?